src/Exception/DeveloperUpdateFailedException.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Copyright 2021 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Exception;
+
+use Apigee\Edge\Exception\ApiException;
+
+/**
+ * This exception is thrown when the developer profile update fails.
+ */
+class DeveloperUpdateFailedException extends ApiException implements ApigeeEdgeExceptionInterface {
```

Comment:
This is not part of the module's public API, make it `final` and properties `private`.
(Unfortunately, we haven't started to use these before the first stable release so every bit of the code can be considered a public API which makes maintenance way more complicated. Although, we can fix our mistakes on new codes.)